pkg/service/generator.go

```diff
@@ -99,6 +106,63 @@ func generateStatefulSetDef(emqx v1beta1.Emqx) *appsv1.StatefulSet {
 	return generateVolume(emqx, sts)
 }
 
+func generateContainerForTelegraf(emqx v1beta1.Emqx, sts *appsv1.StatefulSet) (*corev1.ConfigMap, *appsv1.StatefulSet) {
+
+	telegrafTemplate := emqx.GetTelegrafTemplate()
+	telegrafConfName := fmt.Sprintf("%s-%s", emqx.GetName(), "telegraf-config")
+	logName := fmt.Sprintf("%s-%s", emqx.GetName(), "log")
+
+	containerForTelegraf := corev1.Container{
+		Name:            "telegraf",
+		Image:           telegrafTemplate.Image,
+		ImagePullPolicy: "IfNotPresent",
+		Resources:       emqx.GetTelegrafTemplate().Resources,
+		VolumeMounts: []corev1.VolumeMount{
+			{
+				Name:      telegrafConfName,
```

Comment:
```suggestion
				Name:      "telegraf",
```

Reply:
The `Name` is generated by the `fmt.Sprintf()`, which is related with `EMQX` and used by the `VolumeMouts`,`ConfigMap` and `Volumes`.

Reply:
My bad, misread